conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
```

Comment:
Given that this problem is unique to Windows, I think you want to check that Windows is the platform and then just encoding using UTF-8 otherwise there is some problematic encoding being used.